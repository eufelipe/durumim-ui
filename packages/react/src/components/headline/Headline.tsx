import { ComponentProps, ElementType } from "react";

import { styled } from "../../styles";

export const Headline = styled("h2", {
  fontFamily: "$default",
  lineHeight: "$shorter",
  color: "$gray100",
  margin: 0,

  variants: {
    size: {
      sm: { fontSize: "$xl" },
      md: { fontSize: "$2xl" },
      lg: { fontSize: "$3xl" },
      xl: { fontSize: "$4xl" },
      "2xl": { fontSize: "$5xl" },
    },
  },

  defaultVariants: {
    size: "md",
  },
});

export interface HeadlineProps extends ComponentProps<typeof Headline> {
  as?: ElementType;
}

Headline.displayName = "Headline";
