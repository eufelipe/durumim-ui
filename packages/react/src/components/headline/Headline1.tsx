import { ComponentProps, ElementType } from "react";

import { styled } from "../../styles";

export const Headline1 = styled("h1", {
  fontFamily: "$default",
  fontWeight: "$regular",
  lineHeight: "$shorter",
  color: "$gray100",
  margin: 0,
  size: "5xl",
});

export interface Headline1Props extends ComponentProps<typeof Headline1> {
  as?: ElementType;
}

Headline1.displayName = "Headline1";
