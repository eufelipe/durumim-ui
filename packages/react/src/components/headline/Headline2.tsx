import { ElementType } from "react";

import { CSSProps, css, styled } from "@durumim-ui/core";

const styles = css({
  fontFamily: "$default",
  fontWeight: "$bold",
  lineHeight: "$shorter",
  color: "$gray100",
  margin: 0,
  size: "4xl",
});

const Headline2Component = styled("h2", styles);

export interface Headline2Props extends CSSProps {
  as?: ElementType;
}
export const Headline2: React.FC<Headline2Props> = ({ as, ...props }) => {
  return <Headline2Component as={as} {...props} />;
};

Headline2.displayName = "Headline2";
