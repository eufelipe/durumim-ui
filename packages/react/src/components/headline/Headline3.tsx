import { ElementType } from "react";

import { CSSProps, css, styled } from "@durumim-ui/core";

const styles = css({
  fontFamily: "$default",
  fontWeight: "$regular",
  lineHeight: "$shorter",
  color: "$gray100",
  margin: 0,
  size: "3xl",
});

const Headline3Component = styled("h3", styles);

export interface Headline3Props extends CSSProps {
  as?: ElementType;
}
export const Headline3: React.FC<Headline3Props> = ({ as, ...props }) => {
  return <Headline3Component as={as} {...props} />;
};

Headline3.displayName = "Headline3";
