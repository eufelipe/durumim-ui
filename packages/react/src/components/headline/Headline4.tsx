import { ComponentProps, ElementType } from "react";

import { styled } from "@/styles";

export const Headline4 = styled("h4", {
  fontFamily: "$default",
  fontWeight: "$medium",
  lineHeight: "$shorter",
  color: "$gray100",
  margin: 0,
  size: "2xl",
});

export interface Headline4Props extends ComponentProps<typeof Headline4> {
  as?: ElementType;
}

Headline4.displayName = "Headline4";
