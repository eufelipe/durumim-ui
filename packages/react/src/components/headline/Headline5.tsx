import { ElementType } from "react";

import { CSSProps, css, styled } from "@durumim-ui/core";

const styles = css({
  fontFamily: "$default",
  fontWeight: "$regular",
  lineHeight: "$shorter",
  color: "$gray100",
  margin: 0,
  size: "xl",
});

const Headline5Component = styled("h5", styles);

export interface Headline5Props extends CSSProps {
  as?: ElementType;
}
export const Headline5: React.FC<Headline5Props> = ({ as, ...props }) => {
  return <Headline5Component as={as} {...props} />;
};

Headline5.displayName = "Headline5";
