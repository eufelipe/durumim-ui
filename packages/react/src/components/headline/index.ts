export * from "./Headline1";
export * from "./Headline2";
export * from "./Headline3";
export * from "./Headline4";
export * from "./Headline5";
export * from "./Title";
