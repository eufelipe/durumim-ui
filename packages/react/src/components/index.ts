export * from "./headline";
export * from "./text";
