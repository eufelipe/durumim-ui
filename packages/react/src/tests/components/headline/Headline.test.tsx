import { describe, expect, it } from "vitest";

import { Headline } from "@/components/headline";
import { render, screen } from "@/tests/utils";

describe("<Headline />", () => {
  beforeEach(() => {
    render(
      <Headline data-testid="headline">
        Deixa eu dançar, pro meu corpo ficar odara
      </Headline>
    );
  });

  it("should render the text correctly as expected", () => {
    expect(
      screen.getByText(/Deixa eu dançar, pro meu corpo ficar odara/i)
    ).toBeInTheDocument();
  });

  it("shouldn't render the text correctly", () => {
    expect(
      screen.queryByText(/Minha cara, minha cuca ficar odara/i)
    ).not.toBeInTheDocument();
  });

  it("This should have the color gray100", () => {
    expect(
      screen.getByText(/Deixa eu dançar, pro meu corpo ficar odara/i)
    ).toHaveStyle({
      color: "rgba(184, 191, 204)",
    });
  });

  it("This should have the font size 2xl", () => {
    expect(
      screen.getByText(/Deixa eu dançar, pro meu corpo ficar odara/i)
    ).toHaveStyle({
      "font-size": "1.5em",
    });
  });

  it("should render correctly", () => {
    expect(screen.getByTestId("headline")).toMatchSnapshot();
  });
});
