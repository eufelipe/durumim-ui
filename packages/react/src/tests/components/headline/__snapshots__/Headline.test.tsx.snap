// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Headline /> > should render correctly 1`] = `
<h2
  class="c-eDCyBI c-eDCyBI-kupvTz-size-md"
  data-testid="headline"
>
  Deixa eu dan√ßar, pro meu corpo ficar odara
</h2>
`;
